packages/generator/src/edmx-parser/common/edmx-parser.ts

```diff
@@ -19,10 +21,18 @@ export function parseEntityTypesBase(root): EdmxEntityTypeBase<any>[] {
       PropertyRef: forceArray(e.Key?.PropertyRef)
     },
     NavigationProperty: forceArray(e.NavigationProperty),
-    Property: forceArray(e.Property)
+    Property: forceArray(e.Property),
+    Namespace: e.Namespace
   }));
 }
 
 export function parseEntitySetsBase(root): EdmxEntitySetBase[] {
-  return forceArray(root.EntityContainer.EntitySet);
+  return extractPropertiesFromEntityContainer(root, ec => ec.EntitySet);
+}
+
+export function extractPropertiesFromEntityContainer(
+  root,
+  entityContainerToPropFn
+): any[] {
+  return flat(root.EntityContainer.map(entityContainerToPropFn));
```

Comment:
* Why `extract` not `get`?
* Why properties not property? It is always getting one property, right? (I guess it could do more, but it is never used like this)
* Consider whether passing a string as a second parameter makes it simpler and concentrates the "complexity" in this one place. That way you don't need to know the implementation of this function from the outside.
* Don't we need another forceArray around the property? I think `flat` will fail if it does not get a twodimensional array as input
```suggestion
export function getPropertyFromEntityContainer(
  root,
  entityContainerProperty: string
): any[] {
  return flat(root.EntityContainer.map(ec => forceArray(ec[entityContainerProperty])));
```